api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH706/Moh706LabCohortLibrary.java

```diff
@@ -35,4 +38,182 @@ public CohortDefinition getAllUrineAnalysisGlucoseTestsPositives() {
+        sql.addParameter(new Parameter("endDate", "E    nd Date", Date.class));
+        sql.setQuery("SELECT d.patient_id FROM kenyaemr_etl.etl_patient_demographics d\n" +
+                "                             INNER JOIN kenyaemr_etl.etl_laboratory_extract x ON x.patient_id = d.patient_id\n" +
+                "                             WHERE x.lab_test = 1305 AND x.test_result = 1302 AND date(x.visit_date) BETWEEN :startDate AND :endDate\n" +
+                "                            UNION\n" +
+                "                            SELECT d.patient_id FROM kenyaemr_etl.etl_patient_demographics d\n" +
+                "                             INNER JOIN kenyaemr_etl.etl_laboratory_extract x ON x.patient_id = d.patient_id\n" +
+                "                             WHERE x.lab_test = 856 AND x.test_result IS NOT NULL AND date(x.visit_date) BETWEEN :startDate AND :endDate;"
+
+        );
+        return sql;
+    }
+
+    public CohortDefinition getAllUrineAnalysisProteinsTestsPositives() {
+        SqlCohortDefinition sql = new SqlCohortDefinition();
+        sql.setName("Get urine analysis patients - Proteins");
+        sql.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        sql.addParameter(new Parameter("endDate", "End Date", Date.class));
+        sql.setQuery("SELECT p.patient_id FROM patient p INNER JOIN encounter e ON p.patient_id = e.patient_id INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+                + " WHERE p.voided=0 AND e.voided = 0 AND o.voided = 0 AND o.concept_id IN (1875) AND o.value_coded IN (703,1874,1362,1363,1364,1365) "
+                + " AND e.encounter_datetime BETWEEN :startDate AND :endDate "
+
+        );
+        return sql;
+    }
+
+    public CohortDefinition getAllMalariaTests() {
+        SqlCohortDefinition sql = new SqlCohortDefinition();
+        sql.setName("Get patients with malaria tests");
+        sql.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        sql.addParameter(new Parameter("endDate", "End Date", Date.class));
+        sql.setQuery("SELECT p.patient_id FROM patient p INNER JOIN encounter e ON p.patient_id = e.patient_id INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+                + " WHERE p.voided=0 AND e.voided = 0 AND o.voided = 0 AND o.concept_id IN (32) "
+                + " AND e.encounter_datetime BETWEEN :startDate AND :endDate "
+
+        );
+        return sql;
+    }
+    public CohortDefinition getAllRapidMalariaTests() {
+        SqlCohortDefinition sql = new SqlCohortDefinition();
+        sql.setName("Get patients with rapid malaria tests");
+        sql.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        sql.addParameter(new Parameter("endDate", "End Date", Date.class));
+        sql.setQuery("SELECT p.patient_id FROM patient p INNER JOIN encounter e ON p.patient_id = e.patient_id INNER JOIN obs o ON e.encounter_id = o.encounter_id "
```

Comment:
We don't want to go this direction. Let's consider introducing ETLs to lessen the load.